src/components/CodeSnippet/CodeSnippet.tsx

```diff
@@ -0,0 +1,143 @@
+import classNames from "classnames";
+import PropTypes from "prop-types";
+import React from "react";
+import type { ReactNode } from "react";
+
+export type DropdownOptionPros = {
+  label: string;
+  value: string | number;
+  selected?: boolean;
+  disabled?: boolean;
+};
+
+export type CodeSnippetDropdownProps = {
+  onChange?: (evt: React.SyntheticEvent) => void;
+  options: Array<DropdownOptionPros>;
+};
+
+export const CodeSnippetDropdown = ({
+  options,
+  onChange,
+}: CodeSnippetDropdownProps): JSX.Element => {
+  return (
+    <select className="p-code-snippet__dropdown" onChange={onChange}>
+      {options.map(({ label, value, ...props }) => (
+        <option value={value} key={value || label} {...props}>
+          {label}
+        </option>
+      ))}
+    </select>
+  );
+};
+
+CodeSnippetDropdown.props = {
+  onChange: PropTypes.func,
+  options: PropTypes.arrayOf(
+    PropTypes.shape({
+      label: PropTypes.string.isRequired,
+      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+    })
+  ).isRequired,
+};
+
+export const CodeSnippetBlock = ({
+  code,
+  title,
+  numbered = false,
+  icon = null,
+  isWrapped = false,
+  dropdowns,
+}: CodeSnippetBlockProps): JSX.Element => {
+  let className = "p-code-snippet__block";
+  let numberedCode;
+
+  if (numbered) {
```

Comment:
At the first thought I didn't want to mix icons with numbers in one prop (because numbers aren't really icons, and we have only one icon per block and number per line), but I agree that because numbers and icons are exclusive it kinda makes sense to put them on same prop.

So, I thought that maybe renaming it to `appearance` (one of "numbered" "linuxPrompt" "windowsPrompt" "url") makes sense.